macros/models/_sector/dex/uniswap_compatible_trades.sql

```diff
@@ -8,6 +8,27 @@
 %}
 WITH dexs AS
 (
+{% if blockchain == 'optimism' %}
+    SELECT
+        t.evt_block_number AS block_number,
+        t.evt_block_time AS block_time,
+        t.recipient AS taker,
+        CAST(NULL AS VARBINARY) AS maker,
+        t.amountOut AS token_bought_amount_raw,
+        t.amountIn AS token_sold_amount_raw,
+        null AS amount_usd,
+        t.tokenOut AS token_bought_address,
+        t.tokenIn AS token_sold_address,
+        t.contract_address AS project_contract_address,
+        t.evt_tx_hash AS tx_hash,
+        t.evt_index
+    FROM {{ Pair_evt_Swap }} t
+    {% if is_incremental() %}
+    WHERE {{ incremental_predicate('t.evt_block_time') }}
```

Comment:
Seems pretty clean, but not sure if optimism is an edge case.

As the macro may effect many spell, if the case is not general, we may implement the specific case in its base_trades spell.

Reply:
good point. will update in optimism spell.